game-core/src/test/resources/Total_World_War_Dec1941.xml

```diff
@@ -4,6 +4,66 @@
   <info name="Total World War: December 1941" version="2.7.7.2"/>
   <triplea minimumVersion="1.8"/>
   <diceSides value="12"/>
+  <variableList>
+    <variable name="AntiTankTargets">
+      <element name="germanMech.Infantry"/>
```

Comment:
Is there test code to check for any of these values?

Reply:
Indirectly yes. There are unit tests which parse this XML and test various features with these units. There could be more specific unit tests around individual parts of the XML parsing and what gets created in the data.

Reply:
Well having a completely separate game XML for each unit test would mean you'd have 100s of game XMLs which probably isn't reasonable. While I think trying to make things more isolated would be a good direction, I think you'd have to watch how far you take it and how you structure it. Either way, its outside the scope of this PR.

Reply:
testing these updates well and thoroughly I think is well within the scope of this update. It's hard to be convinced that the current testing is going to find errors. 

Any testing for undefined variables? How about variables that have empty values? Are those detected.

> Well having a completely separate game XML for each unit test would mean you'd have 100s of game XMLs which probably isn't reasonable. 

I'm not suggesting that, but for major parsing features, having an independent test seems not unreasonable. Part of the trick is to split parsing logic from processing logic. A test that verifies we parse a XML correctly and get the right number of elements should scale fine.

Let me know if it would be useful for me to contribute some testing to this branch, I'd be willing to do that.

Reply:
it's also not reasonable for a small change to break every XML test. A small change to add a unit seems will add overhead to update "N" tests. Having worked in environments where one-liners results in a full day of updating tests, it's not a place I want us to be (and to some extent we're there as so many tests are coupled to XML parsing, it's not a good situation).

Reply:
Agree the challenge is that most tests we have today rely on parsing a complete game XML then running various tests over that loaded game. So let's say you went tried to do some parsing optimization changes and made a small error, you'd probably break like 50% of the unit tests.

You are correct that the indirect testing I have is pretty much all positive testing and making sure certain attachments are generated and generated properly not testing if it says generated a extra unintended attachment. It also doesn't test a lot of the "edge" cases around empty or invalid variables.

Right now, I don't think we have any XML parsing specific unit tests and its all just part of many of the others. I think there is value in adding those but its potentially significant effort especially to get even a decent coverage of the various portions of the XML.

Reply:
If we can't get our testing to scale, complex features could be a nail in the efficiency coffin. Notably with a complex feature, it's going to be brittle to make other changes. We're kinda in that state now. Manual testing IMO is an illusion of safety. While better than no testing, manual testing is often called "test-once and never again", but more to the point it rarely has enough coverage. For example rare conditions are likely impossible or never tested manually. It is always a shame when the error handling code has a NPE.

Is it not unreasonable to try and add some simple testing for this feature? Testing should not be so complex, I think part of the issue is we're doing big integration-like tests for all features. A copy-paste of this XML file, minimize it to maybe a hundred lines to have just a delegate and some minimal attributes, should be pretty straight forward, right?

I'm nervous to let this go as I can see the same reasoning being applied ad-infinitum and we'll never get out of the manual testing quagmire.